### ISSUE TITLE
Error vendor/github.com/fsnotify/fsevents/wrap.go:38: constant 36893488147419103231 overflows uint64

### ISSUE THREAD

**Issue by johandalabacka** [fsnotify/fsevents#31]
Before reporting an issue, please ensure you are using the latest release of fsnotify.

### Which version of macOS are you using?

ProductName:    Mac OS X
ProductVersion: 10.13.1
BuildVersion:   17B48

### Please describe the issue that occurred.

While trying to run or buuld the example/main.go (copied to it´s own file) I get

vendor/github.com/fsnotify/fsevents/wrap.go:38: constant 36893488147419103231 overflows uint64

### Are you able to reproduce the issue? Please provide steps to reproduce and a code sample if possible.

1) mkdir test
2) cd test
3) make a file main.go with content of example/main.go
4) dep init
5) go run main.go

**Comment by sroze** [comment]
Looks like the following change makes the trick, but not convinced it's the best idea:
```patch
- const eventIDSinceNow = uint64(C.kFSEventStreamEventIdSinceNow + (1 << 64))
+ const eventIDSinceNow = uint64(C.kFSEventStreamEventIdSinceNow)
```

**Comment by andrewkroh** [comment]
What Go version were you using?

**Comment by johandalabacka** [comment]
I was using 1.9.2 i think. Have not tried your patch :(

2017-12-01 4:48 GMT+01:00 Andrew Kroh <notifications@github.com>:

**Comment by sroze** [comment]
1.9 as well here

**Comment by andrewkroh** [comment]
It only fails on 1.9.2 for me (1.9 and 1.9.1 are good). Looks like golang/go#21708 triggered this error. I'll test the patch above.

**Comment by wadey** [comment]
I've opened a PR here, as part of the fixes for Go 1.10: #34
